src/Legacy/Storage.php

```diff
@@ -2314,29 +2314,34 @@ private function parseWhereParameter($key, $value, $fieldtype = false)
      */
     public function getContentType($contenttypeslug)
     {
-        $contenttypeslug = $this->app['slugify']->slugify($contenttypeslug);
-
         // Return false if empty, can't find it.
         if (empty($contenttypeslug)) {
             return false;
         }
 
+        $contenttype = $this->app['config']->get('contenttypes/' . $contenttypeslug);
+
         // See if we've either given the correct contenttype, or try to find it by name or singular_name.
-        if ($this->app['config']->get('contenttypes/' . $contenttypeslug)) {
-            $contenttype = $this->app['config']->get('contenttypes/' . $contenttypeslug);
-        } else {
-            foreach ($this->app['config']->get('contenttypes') as $key => $ct) {
-                if (isset($ct['slug']) && ($contenttypeslug === $ct['slug'])) {
-                    $contenttype = $ct;
-                    break;
-                }
-                if (isset($ct['singular_slug']) && ($contenttypeslug === $ct['singular_slug'])) {
-                    $contenttype = $ct;
-                    break;
-                }
-                if ($contenttypeslug === $this->app['slugify']->slugify($ct['singular_name']) || $contenttypeslug == $this->app['slugify']->slugify($ct['name'])) {
-                    $contenttype = $ct;
-                    break;
+        if (!$contenttype) {
+            // Also check for the slugified version of the content type
+            $slugifiedContentType = $this->app['slugify']->slugify($contenttypeslug);
+            $contenttype = $this->app['config']->get('contenttypes/' . $slugifiedContentType);
+
+            if (!$contenttype) {
+                foreach ($this->app['config']->get('contenttypes') as $key => $ct) {
+                    if (
+                        (isset($ct['slug']) &&
+                            (($contenttypeslug === $ct['slug']) || ($slugifiedContentType === $ct['slug']))
+                        ) ||
+                        (isset($ct['singular_slug']) &&
+                            (($contenttypeslug === $ct['singular_slug']) || ($slugifiedContentType === $ct['singular_slug']))
+                        ) ||
+                        $slugifiedContentType === $this->app['slugify']->slugify($ct['name']) ||
+                        $slugifiedContentType === $this->app['slugify']->slugify($ct['singular_name'])
+                    ) {
+                        $contenttype = $ct;
+                        break;
+                    }
                 }
```

Comment:
I decided to go with the one liner because it's a tiny bit more efficient, but since we're only dealing with a maximum of three additional checks and four times the number of content types variable assignments I guess it won't matter much. If you guys find this easier to read I'll change it.